src/uu/split/src/split.rs

```diff
@@ -783,6 +807,29 @@ impl Settings {
                 }
             }
         }
+
+        // Make sure that separator is only one UTF8 character (if specified)
+        // defaults to '\n' - newline character
+        // If the same separator (the same value) was used multiple times - `split` should NOT fail
+        // If the separator was used multiple times but with different values (not all values are the same) - `split` should fail
+        let separator = match matches.get_many::<String>(OPT_SEPARATOR) {
+            Some(sep_values) => match sep_values.map(|s| s.to_string()).reduce(|cur, nxt| {
+                if cur == nxt {
+                    nxt
+                } else {
+                    cur + &nxt
+                }
+            }) {
```

Comment:
This is a bit hard to read at the moment. Can we maybe do the reduce on a separate line?